tests/test_authentication.py

```diff
@@ -0,0 +1,138 @@
+mock_session = {'iat': time.mktime(
+    datetime(2011, 6, 21, 10, 0, 0).timetuple())}
+
+mock_openid_config = Mock()
+mock_openid_config.return_value = {'authorization_endpoint':
+                                   'http://dex.example.com:5556/auth',
+                                   'discovery_url':
+                                   'http://dex.example.com:5556',
+                                   'client_id': None}
+
+
+@pytest.yield_fixture
+def mock_views_current_app_config():
+    with mock.patch("app.blueprints.base.views.current_app") as current_app:
+        current_app.config.get.return_value = max_age
+        yield
+
+
+@pytest.yield_fixture
+def mock_oidc_current_app_config():
+    with mock.patch("lib.oidc_old.current_app") as current_app:
+        current_app.config.get.return_value = max_age
+        yield
+
+
+@pytest.yield_fixture
+def mock_notify():
+    with mock.patch("app.blueprints.base.views.notify") as notify:
+        notify.send_email.return_value = ""
+        yield
+
+
+@pytest.fixture
+def test_suit(db_session):
+    suit = Suit(
+        plaintiff=User(
+            email='plaintiff@example.com', name='Plaintiff Test', active=True
+        ),
+        defendant=User(
+            email='defendant@example.com', name='Defendant Test', active=True
+        )
+    )
+    db_session.add(suit)
+    db_session.commit()
+    return suit
+
+
+@pytest.fixture
+def post_accept_suit(client, test_suit):
+    return client.post(url_for('base.accept', suit=test_suit.id),
+                       follow_redirects=False)
+
+
+class WhenTimeSinceLastAuthenticatedIsMoreThanMaxAge(object):
+
+    @patch.dict('app.blueprints.base.views.session', mock_session)
+    @patch('time.time', mock_time_more_than_max_age)
+    def it_returns_false(self, client):
+
+        print(mock_session['iat'])
+        print(time.time())
+
+        assert authenticated_within(max_age) is False
+
+
+class WhenTimeSinceLastAuthenticatedIsLessThanMaxAge(object):
+
+    @patch.dict('app.blueprints.base.views.session', mock_session)
+    @patch('time.time', mock_time_less_than_max_age)
+    def it_returns_true(self, client):
+
+        print(mock_session['iat'])
+        print(time.time())
+
+        assert authenticated_within(max_age) is True
+
+
+class WhenAcceptingASuitWithinAuthenticatedTime(object):
+
+    @patch('time.time', mock_time_less_than_max_age)
+    def it_redirects_to_admin(self, test_admin_user, client, test_suit,
+                              mock_notify, mock_views_current_app_config):
+
+        with client.session_transaction() as session:
+            session['user_id'] = test_admin_user.id
+            session['_fresh'] = False
+            session['iat'] = mock_session['iat']
+
+        response = client.post(url_for('base.accept', suit=test_suit.id))
+
+        print(response.headers["Location"])
+        assert response.status_code == 302
+        assert ("/admin" in
+                response.headers["Location"])
```

Comment:
is this line break necessary? also, the brackets around the expression are not necessary